src/extension/importer/rest/src/main/java/org/geoserver/importer/rest/converters/ImportJSONReader.java

```diff
@@ -420,7 +421,16 @@ FileData file(JSONObject json) throws IOException {
         if (json.has("file")) {
             // TODO: find out if spatial or not
             String file = json.getString("file");
-            return FileData.createFromFile(new File(file));
+            FileData importFileData = FileData.createFromFile(new File(file));
+            // check if charsetEncoding configured (Geos-9073)
+            if (json.has("charsetEncoding")) {
+                // check if charsetEncoding is supported
+                // dont upload a charset which not supported
+                Charset.isSupported(json.getString("charsetEncoding"));
```

Comment:
I would have expected an if (Charset.isSupported(...)).
Is this throwing an exception if the encoding is not supported?
From the docs the exception should be thrown only with null or invalid name.

Reply:
understood, will put an inverted if clause on it.